### PR TITLE
sig-storage: cosi: prow: enable cloud build

### PR DESCRIPTION
**Pull request by BlaineEXE**
Enable cloud builds for sig-storage COSI [kep-1979].

### CODE REVIEW

File: config/jobs/kubernetes-sigs/container-object-storage-interface/container-object-storage-interface-controller.yaml [hunk 1]
```diff
@@ -1,49 +1,78 @@
...
+      path_alias: sigs.k8s.io/container-object-storage-interface-controller
+      annotations:
+        testgrid-dashboards: sig-storage-container-object-storage-interface-controller
+        testgrid-tab-name: build
+        description: Build test in container-object-storage-interface-controller repo.
+      spec:
+        containers:
+          - image: gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:latest
+            command:
+              - make
```
Comment (wlan0) [inline config/jobs/kubernetes-sigs/container-object-storage-interface/container-object-storage-interface-controller.yaml:16]: Is there a mix of tabs and spaces?  Please make sure it’s just one type of spacing and that it matches rest of the yaml files in this repo.

File: config/jobs/kubernetes-sigs/container-object-storage-interface/container-object-storage-interface-controller.yaml [hunk 2]
```diff
@@ -1,49 +1,77 @@
...
+    # based on https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/README.md#prow-config-template
+    - name: post-container-object-storage-interface-controller-push-images
+      cluster: k8s-infra-prow-build-trusted
+      annotations:
+        # have to report image builds to sig-storage-image-build: https://testgrid.k8s.io/sig-storage-image-build
+        testgrid-dashboards: sig-storage-image-build
+      decorate: true
+      branches:
+        #<!-- disable image builds from master temporarily --># - ^master$
+        - ^canary$ # use canary branch for testing image build itself
```
Comment (BlaineEXE) [inline config/jobs/kubernetes-sigs/container-object-storage-interface/container-object-storage-interface-controller.yaml:64]: builds are only enabled on canary while we are testing

**Comment by BlaineEXE** [comment]
I think I finally have a working idea of how the sig build tooling is/should be put together and that this item is auto-built for all sig-storage projects via this script: https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-sig-storage.sh

**Pull request closed without merging**